frontend/public/components/cluster-settings/remoteauth-idp-form.tsx

```diff
@@ -0,0 +1,228 @@
+    if (_.isEmpty(this.state.keyFileContent) !== _.isEmpty(this.state.certFileContent)) {
+      this.setState({errorMessage: 'Values for certFile and keyFile should both be either excluded or provided.'});
+      return;
+    }
+    // Clear any previous errors.
+    this.setState({errorMessage: ''});
+    this.getOAuthResource().then((oauth: OAuthKind) => {
+      const promises = [
+        this.createCertSecret(),
+        this.createKeySecret(),
+        this.createCAConfigMap(),
+      ];
+
+      Promise.all(promises).then(([certSecret, keySecret, configMap]) => {
+        const caName = configMap ? configMap.metadata.name : '';
+        const certSecretName = certSecret ? certSecret.metadata.name: '';
+        const keySecretName = keySecret ? keySecret.metadata.name: '';
+        return this.addRemoteAuthIDP(oauth, certSecretName, keySecretName, caName);
+      }).then(redirectToOAuthPage);
+    });
+  };
+
+  nameChanged: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    this.setState({name: event.currentTarget.value});
+  };
+
+  urlChanged: React.ReactEventHandler<HTMLInputElement> = (event) => {
+    this.setState({url: event.currentTarget.value});
+  };
+
+  caFileChanged = (caFileContent: string) => {
+    this.setState({caFileContent});
+  };
+
+  certFileChanged = (certFileContent: string) => {
+    this.setState({certFileContent});
+  };
+
+  keyFileChanged = (keyFileContent: string) => {
+    this.setState({keyFileContent});
+  };
+
+  render() {
+    const { name, url, caFileContent, certFileContent, keyFileContent } = this.state;
+    const title = 'Add Identity Provider: Remote Authentication';
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form onSubmit={this.submit} name="form" className="co-m-pane__body-group co-m-pane__form">
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">
+          Basic authentication is a generic backend integration mechanism that allows users to authenticate with credentials validated against a remote identity provider.
+        </p>
+        <IDPNameInput value={name} onChange={this.nameChanged} />
+        <div className="form-group">
+          <label className="control-label co-required" htmlFor="url">URL</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.urlChanged}
+            value={url}
+            id="url"
+            aria-describedby="idp-url-help"
+            required />
+          <p className="help-block" id="idp-url-help">
+            The OAuth server base URL.
+          </p>
+        </div>
+        <IDPCAFileInput value={caFileContent} onChange={this.caFileChanged} />
+        <div className="form-group">
+          <DroppableFileInput
+            onChange={this.certFileChanged}
+            inputFileData={certFileContent}
+            id="cert-file-input"
+            label="certFile"
```

Comment:
```suggestion
            label="Certificate"
```

We should add a help block:

> PEM-encoded TLS client certificate to present when connecting to the server.

Reply:
@spadgett I was using the designs here: https://docs.google.com/document/d/15MUwbd92RsZp5NMB1zyBzUwDBGupz90wqrku0iyC76c/edit#heading=h.arp1o27fsss9
I'll adjust based on your comments.